### PR TITLE
refactor: remove dependency of llmq/chainlocks, llmq/quorum_block_processor, ehf_signals on PeerManager

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
It reduces circular dependencies and simplify code.

## What was done?
Removed circular dependency over PeerManager for llmq::QuorumBlockProcessor, llmq::CEhfSignals, llmq::ChainLocks, and several extra useful refactorings: see commits.

## How Has This Been Tested?
Run `test/lint/lint-circular-dependencies.sh`

## Breaking Changes
N/A

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: src/llmq/dkgsessionhandler.cpp [hunk 1]
```diff
@@ -612,7 +612,10 @@ void CDKGSessionHandler::HandleDKGRound()
 
     auto finalCommitments = curSession->FinalizeCommitments();
     for (const auto& fqc : finalCommitments) {
-        quorumBlockProcessor.AddMineableCommitmentAndRelay(fqc);
+        auto inv_opt = quorumBlockProcessor.AddMineableCommitment(fqc);
+        if (inv_opt.has_value()) {
```
Comment (PastaPastaPasta) [inline src/llmq/dkgsessionhandler.cpp:616]: nit: could combine

File: src/llmq/ehf_signals.cpp [hunk 2]
```diff
@@ -108,11 +108,11 @@ MessageProcessingResult CEHFSignalsHandler::HandleNewRecoveredSig(const CRecover
     }
 
     const auto ehfSignals = mnhfman.GetSignalsStage(WITH_LOCK(cs_main, return chainstate.m_chain.Tip()));
-    MNHFTxPayload mnhfPayload;
     for (const auto& deployment : Params().GetConsensus().vDeployments) {
         // skip deployments that do not use dip0023 or that have already been mined
         if (!deployment.useEHF || ehfSignals.find(deployment.bit) != ehfSignals.end()) continue;
 
+        MNHFTxPayload mnhfPayload;
```
Comment (PastaPastaPasta) [inline src/llmq/ehf_signals.cpp:114]: 595c4cd88b0935e17ffcdcd096a180d444b48cc0 this commit shouldn't be included here? If it's a fix? I'm confused by it

**Review by PastaPastaPasta — COMMENTED** [review]
generally LGTM, don't **love** 5e489bc9c6ba33c8caf42649773f68e4a82b1651 feels like it makes struct MessageProcessingResult too complicated

**Comment by knst** [comment]
> @[knst](https://github.com/knst) knst [force-pushed](https://github.com/dashpay/dash/compare/5e489bc9c6ba33c8caf42649773f68e4a82b1651..5739e777d4133ef7dd0b92e0742c02023825ce26) the fix-circular-net-processing-p3 branch from[ 5e489bc ](https://github.com/dashpay/dash/commit/5e489bc9c6ba33c8caf42649773f68e4a82b1651)to 5739e77

1st force-push to resolve conflict

> @[knst](https://github.com/knst) knst [force-pushed](https://github.com/dashpay/dash/compare/5739e777d4133ef7dd0b92e0742c02023825ce26..32683b3bd91513b269d33343357f3035e5394839) the fix-circular-net-processing-p3 branch from[ 5739e77 ](https://github.com/dashpay/dash/commit/5739e777d4133ef7dd0b92e0742c02023825ce26)to 32683b3

2nd force-push to address comments

**Comment by UdjinM6** [comment]
build fails

**Review by UdjinM6 — DISMISSED** [review]
utACK d0f17788fc99804e34c5cc7b0274a38a6f8105d8

File: src/llmq/chainlocks.cpp [hunk 3]
```diff
@@ -525,10 +525,10 @@ void CChainLocksHandler::EnforceBestChainLock()
     uiInterface.NotifyChainLock(clsig->getBlockHash().ToString(), clsig->getHeight());
 }
 
-void CChainLocksHandler::HandleNewRecoveredSig(const llmq::CRecoveredSig& recoveredSig)
+MessageProcessingResult CChainLocksHandler::HandleNewRecoveredSig(const llmq::CRecoveredSig& recoveredSig)
```
Comment (PastaPastaPasta) [inline src/llmq/chainlocks.cpp:504]: I'm confused, this method doesn't doesn't return any error values?

File: src/llmq/ehf_signals.cpp [hunk 4]
```diff
@@ -94,15 +94,15 @@ void CEHFSignalsHandler::trySignEHFSignal(int bit, const CBlockIndex* const pind
     sigman.AsyncSignIfMember(llmqType, shareman, requestId, msgHash, quorum->qc->quorumHash, false, true);
 }
 
-void CEHFSignalsHandler::HandleNewRecoveredSig(const CRecoveredSig& recoveredSig)
+MessageProcessingResult CEHFSignalsHandler::HandleNewRecoveredSig(const CRecoveredSig& recoveredSig)
```
Comment (PastaPastaPasta) [inline src/llmq/ehf_signals.cpp:94]: same?

File: src/llmq/chainlocks.cpp [hunk 5]
```diff
@@ -98,31 +96,11 @@ CChainLockSig CChainLocksHandler::GetBestChainLock() const
...
-        return {};
-    }
-
-    if (msg_type == NetMsgType::CLSIG) {
-        CChainLockSig clsig;
-        vRecv >> clsig;
-
-        return ProcessNewChainLock(pfrom.GetId(), clsig, ::SerializeHash(clsig));
-    }
-    return {};
```
Comment (PastaPastaPasta) [inline src/llmq/chainlocks.cpp:108]: why drop this?

Comment (PastaPastaPasta) [reply src/llmq/chainlocks.cpp:108]: nvm

File: src/protocol.h [hunk 6]
```diff
@@ -585,6 +585,7 @@ struct MessageProcessingResult
 {
     std::optional<MisbehavingError> m_error;
     std::optional<CInv> m_inventory;
+    std::optional<CInv> m_to_erase;
```
Comment (PastaPastaPasta) [inline src/protocol.h]: confused what the difference is between these? maybe document this API a bit?

File: src/llmq/chainlocks.cpp (hunk 3 above)
Comment (knst) [reply src/llmq/chainlocks.cpp:504]: `HandleNewRecoveredSig` is override of interface method
```
[[nodiscard]] virtual MessageProcessingResult HandleNewRecoveredSig(const CRecoveredSig& recoveredSig) = 0;
```

which belongs to `CRecoveredSigsListener`.
Some classes which implements `CRecoveredSigsListener` has errors to return, some - doesn't. See other commits, for example: https://github.com/dashpay/dash/pull/6292/files/#diff-f7c0d20797337d77a9cfbf40c4f627ec68ed5a12e2f1193fa91c32cf285d9e34R526 - returns MisbehavingError when ProcessNewChainlock or https://github.com/dashpay/dash/pull/6292/files/#diff-93616b1f609ab7dc32d07bcdd3de3611d59bb285cbd6dba5a3bc5b4c70e6c867R131 - return TX

**Review by UdjinM6 — APPROVED** [review]
utACK c77216ea80f47af369c27fe8a18cfd2b88212a65

**Review by PastaPastaPasta — COMMENTED** [review]
utACK c77216ea80f47af369c27fe8a18cfd2b88212a65

**Pull request merged**